docs/src/kb/git/git-setting-up-git-and-github-UFSWbB.rst

```diff
@@ -0,0 +1,172 @@
+
+.. code-block:: bash
+
+    git config --global user.name "Your Name"
+    git config --global user.email "youremail@example.com"
+
+GitHub recently changed the default branch on new repositories from `master` to `main`. Change the default branch for Git using this command:
+
+.. code-block:: bash
+
+    git config --global init.defaultBranch main
+
+You’ll also likely want to set your default branch reconciliation behavior to merging.
+
+.. code-block:: bash
+
+    git config --global pull.rebase false
+
+To verify that things are working properly, enter these commands and verify whether the output matches your name and email address.
+
+.. code-block:: bash
+
+    git config --get user.name
+    git config --get user.email
+
+For macOS Users
+~~~~~~~~~~~~~~~
+
+Run these two commands to tell Git to ignore `.DS_Store` files, which are automatically created when you use Finder to look into a folder.
+
+.. code-block:: bash
+
+    echo .DS_Store >> ~/.gitignore_global
+    git config --global core.excludesfile ~/.gitignore_global
+
+Step 4: Connect to GitHub with SSH
+----------------------------------
+
+To securely connect to GitHub from your computer, it's best to use SSH (Secure Shell). This allows you to push and pull code without having to enter your username and password every time.
+
+1.  **Generate a new SSH key**
+
+    First, check if you already have an Ed25519 SSH key.
+
+    .. code-block:: bash
+
+        ls ~/.ssh/id_ed25519.pub
+
+    If you see a "No such file or directory" message, you need to create one.
+
+    .. code-block:: bash
+
+        ssh-keygen -t ed25519
+
+    When you're prompted to "Enter a file in which to save the key," you can press Enter to accept the default file location. You can optionally enter a passphrase for extra security.
+
+2.  **Add your SSH key to your GitHub account**
+
+    First, you need to copy your public SSH key. You can display it in the terminal with this command:
+
+    .. code-block:: bash
+
+        cat ~/.ssh/id_ed25519.pub
+
+    Copy the entire output, which starts with `ssh-ed25519` and ends with your email address.
+
+    Now, go to your GitHub account settings:
+
+    - Click on your profile picture in the top-right corner and select **Settings**.
+    - In the left sidebar, click on **SSH and GPG keys**.
+    - Click **New SSH key** or **Add SSH key**.
+    - Give your key a title (e.g., "My Laptop") and paste the key into the "Key" field.
+    - Click **Add SSH key**.
+
+3.  **Test your SSH connection**
+
+    Follow the `GitHub directions for testing your SSH connection <https://docs.github.com/en/authentication/connecting-to-github-with-ssh/testing-your-ssh-connection>`_.
+
+    You should see a message like this:
+
+    .. code-block:: text
+
+        Hi username! You’ve successfully authenticated, but GitHub does not provide shell access.
+
+You're all set! You can now start using Git and GitHub to manage your projects.
```

Comment:
On cluster environment (if your system administrator does not allow port 22), you need this

https://docs.github.com/en/authentication/troubleshooting-ssh/using-ssh-over-the-https-port